### ISSUE TITLE
Having a icon / a way to know which type of variable the auto-completion is showing.

### ISSUE THREAD

**Issue by WAYxTREM** [godotengine/godot-proposals#3291]
### Describe the project you are working on

I'm working on a game inspired of Kirby Returns To Dreamlands, it's a 2.5D platformer game with a very fantasy style and with enemies (in multiple levels) and bosses and to defends yourself you will have to use super-powers like (laser , explosive , lava...) .I'm planning on publishing it on Nintendo Switch with the publisher gotm.io

### Describe the problem or limitation you are having in your project

The auto-completion can be very confusing because i don't know the kind of variable type it is requesting me to use.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I think Godot engine could implements little icon next to the variable name to show the type. It would be useful to see which type it is to be more efficient and to prevent from getting errors.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The way it will work is having a little icon , instead of having the .P icon , we could have a icon little .Bool or .Float
I think there is something like that in Godot's Visual Scripting solution and that's could be cool to have this on the code edtior.
![image](https://user-images.githubusercontent.com/74776686/133002325-aee2cb85-0240-409f-ba28-d62b3d0e9a2c.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think it can. I didn't tried tho.

### Is there a reason why this should be core and not an add-on in the asset library?

Because this is a very important features. A lot , if not all of them show which type of variable it is to make the user more efficent and to prevent his code from having impossible convertions.

**Comment by Calinou** [comment]
This is how Visual Studio Code displays type information in the autocompletion box (by also displaying the function signature on the right):

![image](https://user-images.githubusercontent.com/180032/133002498-d26f25a0-36ed-4f0f-a9ca-d6f910137fdd.png)

**Comment by LightningAA** [comment]
Somewhat related: https://github.com/godotengine/godot-proposals/issues/99

**Comment by WAYxTREM** [comment]
> 
> 
> For reference, this is how Visual Studio Code displays type information in the autocompletion box (by also displaying the function signature on the right):
> 
> ![image](https://user-images.githubusercontent.com/180032/133002498-d26f25a0-36ed-4f0f-a9ca-d6f910137fdd.png)
> 
> It also works for properties:
> 
> ![image](https://user-images.githubusercontent.com/180032/133002512-a74a1c54-8523-4615-be32-adc8bf796430.png)
> 
> I would prefer this as it won't be limited to primitive (non-object) types.

Yes, but at the in the little rectangle it shows you the type of variable it will uses , Godot doesn't display that (ex: erand48 is a double and N is a List<String>)

**Comment by EricEzaM** [comment]
Related: https://github.com/godotengine/godot/pull/38449

This PR of mine was closed as the Text/Code editor were in a state of flux at the time. Now that the API has settled a bit, it might be a good time to revisit.
